lib/auth0_client.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+require 'jwt'
+require 'net/http'
+
+class Auth0Client
+  Error = Struct.new(:message, :status)
+  Response = Struct.new(:decoded_token, :error)
+
+  def self.domain_url
+    "https://#{Rails.application.config.x.auth0.domain}/"
+  end
+
+  def self.decode_token(token, jwks_hash)
+    JWT.decode(token, nil, true, {
+                 algorithm: 'RS256',
+                 iss: domain_url,
+                 verify_iss: true,
+                 aud: Rails.application.config.x.auth0.audience,
+                 verify_aud: true,
+                 jwks: { keys: jwks_hash[:keys] }
+               })
+  end
+
+  def self.jwks
```

Comment:
I think I'd probably rename this to something that more clearly indicates that it's going to go fetch a file from the internet. `self.jwks` sounds like it's just going to return some hardcoded value, kind of like in instance variable in most languages, but it's doing a lot more than just that.

Maybe `fetch_jwks` or `get_jwks`?

Reply:
Also, this might be more of a personal style preference, but I feel like I'd prefer most of the methods in this class to all be merged together a single one. As a reader, when you see many methods, then you tend to think "oh, what is the API of this method, and who is expected to call it?" But if these aren't actually being used in a lot of places, they could really just be replaced with local variables, where you know that the variables do not get used outside of the method.

Reply:
Thank you, that is helpful feedback. I've consolidated some of the methods in the class and renamed jwks to the more descriptive "fetch_jwks"